docs/src/main/asciidoc/_admin-guide/subsystem-configuration/MicroProfile_Config_SmallRye.adoc

```diff
@@ -145,6 +145,54 @@ This results in the XML configuration:
 
 All properties from the `ConfigSource`s provided by this `ConfigSourceProvider` will be available to any WildFly deployment.
 
+=== ConfigSourceProvider from Root directory
+You can create a `ConfigSourceProvider` for a root directory by creating a `config-source-root` resource pointing to a
+root directory. Top level directories within this root directory each become an individual `ConfigSource` reading from a
+directory similar to what we saw earlier in <<configsource-from-directory, ConfigSource from Directory>>.
+
+This is especially useful when running on OpenShift where `ConfigMap` and `ServiceBinding` instances get mapped under
+a common known location. For example if there are two `ConfigMap` instances (for this example, we will call them
+`map-a`, and `map-b`) used by your application pod, they will
+each get mapped under `/etc/config`. So you would have `/etc/config/map-a` and `/etc/config/map-b` directories. Each of
+these directories will have files where the file name is the name of the property and the file content is the value of
+the property, like we saw earlier.
+
+We can thus simply run the following CLI command to pick up all these child directories as a `ConfigSource` each:
+[source,options="nowrap"]
+----
+/subsystem=microprofile-config-smallrye/config-source-root=config-maps:add(path=/etc/config)
+----
+
+This results in the XML configuration:
+
+[source,xml,options="nowrap"]
+----
+<subsystem xmlns="urn:wildfly:microprofile-config-smallrye:2.0">
+    <config-source-root name="config-maps" path="/etc/config"/>
+</subsystem>
+----
+Still assuming the `/etc/config` directory contains the `map-a` and `map-b` directories we are using for this examples,
+the above is analogous to doing:
+----
+/subsystem=microprofile-config-smallrye/config-source=file-props:add(dir={path=/etc/config/map-a})
+/subsystem=microprofile-config-smallrye/config-source=file-props:add(dir={path=/etc/config/map-b})
+----
+Using `config-source-root` removes the need to know the exact names of each entry under the common parent directory.
+
+The situation where two `ConfigSource` entries under the same root both contain the same property should be avoided. However, to make this situation deterministic, the directories representing each `ConfigSource` are sorted by their name according to standard Java sorting rules before doing the lookup of values. To make this more concrete, if we have the following entries:
+* `/etc/config/map-a/num.max` contains `100`
+* `/etc/config/map-b/num.max` contains `200`
+
+Since `map-a` will come before `map-b` after sorting, in the following scenario `200` (coming from `map-a`) will be injected
```

Comment:
Your example is incorrect:
> Since `map-a` will come before `map-b` after sorting, in the following scenario `200` (coming from `map-a`) will be injected

should be

> Since `map-a` will come before `map-b` after sorting, in the following scenario `100` (coming from `map-a`) will be injected

this could be avoided by explicitly setting the ordinal in each top-level dir according to https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#manually-defining-the-ordinal-of-a-built-in-configsource


* `/etc/config/map-a/config_ordinal -> 101`
* `/etc/config/map-b/config_ordinal -> 102`
-> will always look at map-b first so that `num.max` is resolved to `200`